hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractCompletableFuture.java

```diff
@@ -110,25 +163,71 @@ public V get() throws InterruptedException, ExecutionException {
         }
     }
 
+    @Override
+    public V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+        final long deadlineTimeMillis = System.currentTimeMillis() + unit.toMillis(timeout);
+        long millisToWait;
+        for (; ; ) {
+            Object currentState = this.state;
+
+            if (isCancelledState(currentState)) {
+                throw new CancellationException();
+            }
+            if (isDoneState(currentState)) {
+                return getResult();
+            }
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            millisToWait = deadlineTimeMillis - System.currentTimeMillis();
+            if (millisToWait <= 0) {
+                throw new TimeoutException();
+            }
+
+            synchronized (this) {
+                if (!isDoneState(this.state)) {
```

Comment:
The idea was to have an analogous naming convention to the Future's `isDone()` and `isCancelled()` one, thus we named it `isDoneState()` and `isCancelledState()`.

Introducing another word like `final` doesn't help much here IMHO, because somebody still has to know if `finalState` encompasses `cancelled` or not.
I would call the method `isDoneOrCancelled()` - then it's obvious what it means.

WDYT?